presto-main/src/main/java/com/facebook/presto/execution/scheduler/ConsistentHashingNodeProvider.java

```diff
@@ -56,14 +58,21 @@ public ConsistentHashingNodeProvider(NavigableMap<Integer, InternalNode> candida
     public List<HostAddress> get(String key, int count)
     {
         ImmutableList.Builder<HostAddress> nodes = ImmutableList.builder();
-        for (int i = 0; i < count; i++) {
-            int hashKey = HASH_FUNCTION.hashString(format("%s%d", key, i), UTF_8).asInt();
+        Set<HostAddress> unique = new HashSet<>();
+        int i = 0;
+        while (unique.size() < count) {
```

Comment:
Could this be an infinite loop? e.g., when num of workers is < count? Add a test for this case?

It is also possible that this will take a lot of iterations to find next unique candidate if the hash function happens to return a visited virtual node for a lot of times. 
How about searching it in the candidates map to find the next unique candidate instead of relying on a random function? Specifically, if we find a duplicate node, we get its iterator/pointer in the map and check the following nodes according to their order in the map until we find a new node or have exhausted all node.

Reply:
Hmm, I think the assumption here is that count is small. 2 is used right now. But I agree we should add a check. I think in any realistic scenario the probability of continuously hitting the same virtual nodes is pretty low but that's also assuming count is small. Next physical node on the ring seems better.